dts/bindings/phy/st,stm32u5-otghs-phy.yaml

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2024 Marcin Niestroj
+# Copyright (c) 2024 Meta
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+    This binding is to be used by the STM32U5xx transceivers which are built-in
+    with USB HS PHY IP and a configurable HSE clock source.
+
+compatible: "st,stm32u5-otghs-phy"
+
+include: phy-controller.yaml
+
+properties:
+  "#phy-cells":
+    const: 0
+
+  clocks:
+    required: true
+    description: |
+      Supported configurations:
+
+        /* HSE */
+        clocks = <&rcc STM32_CLOCK(AHB2, 15U)>,
+                 <&rcc STM32_SRC_HSE OTGHS_SEL(0)>;
+
+        /* HSE/2 */
+        clocks = <&rcc STM32_CLOCK(AHB2, 15U)>,
+                 <&rcc STM32_SRC_HSE OTGHS_SEL(2)>;
+
+        /* PLL1_P_CK */
+        clocks = <&rcc STM32_CLOCK(AHB2, 15U)>,
+                 <&rcc STM32_SRC_PLL1_P OTGHS_SEL(1)>;
+
+        /* PLL1_P_CK/2 */
+        clocks = <&rcc STM32_CLOCK(AHB2, 15U)>,
+                 <&rcc STM32_SRC_PLL1_P OTGHS_SEL(3)>;
```

Comment:
I would actually prefer we introduce a new `STM32_SRC_PLL1_P_DIV_2` source to avoid breaking `clock_control_get_rate()` api behavior (with your proposal it may not return the right rate anymore).

To make it reusable, we can think of adding a trick on the u32 value such as :
```
                 <&rcc (STM32_SRC_PLL1_P | DIV_2) OTGHS_SEL(3)>;
```
With the DIV_2 being extracted in `clock_control_get_rate()` and included in rate computation before returning (to make it functional with similar HSE/2, HSI48/2, HSI/256, MSIK/1024 and MSIK/4)

Reply:
Great feedback!

I've implemented something along those lines, which allows to use:

```
<&rcc (STM32_SRC_PLL1_P | STM32_CLOCK_DIV(2)) OTGHS_SEL(3)>;
```